rest/src/main/kotlin/builder/components/CompositeComponentBuilder.kt

```diff
@@ -0,0 +1,34 @@
+package dev.kord.rest.builder.components
+
+import dev.kord.common.annotation.KordPreview
+import dev.kord.common.entity.ComponentType
+import dev.kord.common.entity.DiscordComponent
+import dev.kord.common.entity.optional.optional
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.InvocationKind
+import kotlin.contracts.contract
+
+@KordPreview
+class CompositeComponentBuilder {
+    var components: MutableList<DiscordComponent> = mutableListOf()
```

Comment:
Consider making this a val instead. Certain operator functions don't play nice with vars of mutable lists.